go/plugins/vertexai/vertexai.go

```diff
@@ -70,7 +71,19 @@ func Init(ctx context.Context, projectID, location string) error {
 	if state.initted {
 		panic("vertexai.Init already called")
 	}
+	if projectID == "" {
+		projectID = os.Getenv("GCLOUD_PROJECT")
```

Comment:
for the record/followup
```go
import 	"cloud.google.com/go/auth/credentials"


	creds, err := credentials.DetectDefault(&credentials.DetectOptions{})
	if err != nil {
		return err
	}
	projectID, err := creds.QuotaProjectID(ctx)
	if err != nil {
		return err
	}
```

Reply:
I was expecting `creds.ProjectID(ctx)` to work... but doesn't for some reason.

Reply:
I looked at this, it only handles ADC: https://pkg.go.dev/cloud.google.com/go/auth/credentials#DetectDefault

There's no client library that does everything the Node.js google-auth-library does AFAICT.

Reply:
we also only do ADC in nodejs